### ISSUE TITLE
LocationActivity bulletproof

### ISSUE THREAD

**Issue by ensozos** [paddybc/TearfundDisasterResponse#9]
``` java
locationManager.removeUpdates(locationListener);
```

We must check to see if permision is available( permisions can be rejected by user).
Same thing below

``` java
locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, (long) 0, (float) 0, locationListener);
```

**Comment by sampeckham** [comment]
Yep! Had a few issues with that today, either taking quite a while, or never getting location. Made me wonder if we needed to have a permissions set on installing. Same goes for camera access, right?

**Comment by UsAndRufus** [comment]
Sam, if you can't get location to work try swapping NETWORK_PROVIDER to GPS_PROVIDER. Also, any permissions set in the manifest have to be agreed to when installing via Google Play - they are not being requested from the user here because the app is installed via USB. But yes, we should probably do some permission checking.

**Comment by HappySmilyGuy** [comment]
...  does it currently try and get location via GPS then if it fails via the network provider? If not I'll try and change it to do that.
